### DISCUSSION TITLE
Allow addition/subtraction assignment operators when deriving record using `with`

### DISCUSSION THREAD

**Discussion by WhitWaldo** [dotnet/csharplang#7356] [Language Ideas]
This is a relatively minor feature request that I'd like to submit simply for consistency of language functionality across the newer records implementation. When creating a derived record using `with`, a constant or variable value must be assigned to the named property. For the sake of conciseness, I'd like to request that this assignment also support addition and subtraction operators.

Today, when looking to increment the value of a property in a record, I have to do something like the following:

```cs
public void DoStuff()
{
  var thing = new MyRecord("Apple", 5);
  var newThing = thing with {
    Count = thing.Count + 1
  }
}
```

This isn't terrible, but it does mean I repeat myself - I'm deriving this new object from `thing` and then referencing this `thing` to get the Count value. This leaves an opportunity for a typo and inadvertently pulling the value from an unintended other source. Rather, because I can use addition and subtraction assignment operators during a more traditional assignment, I'd like to request that I can do the same here:

```cs
public void DoStuff()
{
  var thing = new Record("Apple", 5);
  var newThing = thing with {
    Count += 1
  }; //Count = 6
}
```

This eliminates any typo opportunity, makes the syntax ever so slightly more concise and continues to bring parity between traditional assignment approaches and the new functionality available with records.

Thank you for your consideration!

**Comment by CyrusNajmabadi** [comment]
Dupe of https://github.com/dotnet/csharplang/issues/5176.  Thanks! :)
